### PR TITLE
William/no modal manager

### PR DESCRIPTION
**Pull request by swansontec**
### Changed

- Add a submit button to the following modals:
  - Fio address registration entry
  - Fio domain registration entry
  - Paste private key (scan scene)
  - Upload logs notes
- Upgrade the following modals to use modern theming:
  - Wallet name entry
  - Custom plugin URI
  - Promotion code entry
  - Password validation (view wallet keys, unlock settings)

#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [x] Tested on iOS Tablet
- [x] Tested on small Android
- [ ] n/a

### CODE REVIEW

**Comment by swansontec** [comment]
Screenshot_20210805-151054_Edge
Screenshot_20210805-151241_Edge

**Comment by swansontec** [comment]
Text modals look like this:

Screenshot_20210803-151124_Edge

**Comment by swansontec** [comment]
Built on top of #2819

File: src/components/modals/TextInputModal.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    onSubmit(text).then(
+      result => {
+        if (result === true) {
+          bridge.resolve(text)
+        } else if (result === false) {
+          setSpinning(false)
+        } else {
+          setSpinning(false)
+          setErrorMessage(result)
+        }
```
Comment (thehobbit85) [inline src/components/modals/TextInputModal.js:71]: Flip around

**Review by thehobbit85 — CHANGES_REQUESTED** [review]

File: src/components/modals/TextInputModal.js (hunk 1 above)
Comment (swansontec) [reply src/components/modals/TextInputModal.js]: Yeah, much nicer that way!

**Pull request merged**